state-sync/diem-data-client/src/lib.rs

```diff
@@ -130,38 +133,87 @@ pub trait DiemDataClient {
 /// the proof failed verification). This can be done using the
 /// `notify_bad_response()` API call above.
 #[derive(Clone, Debug)]
-pub struct DataClientResponse {
-    pub response_id: u64,
-    pub response_payload: DataClientPayload,
+pub struct Response<T> {
+    id: u64,
+    payload: T,
 }
 
-/// The payload returned in a Data Client response.
+impl<T> Response<T> {
+    pub fn new(id: u64, payload: T) -> Self {
+        Self { id, payload }
+    }
+
+    pub fn id(&self) -> u64 {
```

Comment:
nit: I don't really see the need for all these accessor methods. Why not just make the fields public?